Tasks/Common/azure-arm-rest/AzureResourceFilter.ts

```diff
@@ -0,0 +1,33 @@
+import msRestAzure = require('./azure-arm-common');
+import webClient = require('./webClient');
+import azureServiceClient = require('./AzureServiceClient');
+import tl = require('vsts-task-lib/task');
+import Q = require('q');
+import { AzureResourceFilterManagementClient } from './azure-arm-resource-filter';
+import { AzureEndpoint } from './azureModels';
+
+export class AzureResourceFilterController {
```

Comment:
why do you need this wrapper class. If it is for callback to promise conversion - you can directly use promises in the client class itself

Reply:
@sachinma , this wrapper class is created as the logic required to find resources is common for Manage and deploy task. Resolving the promise here.